### PR TITLE
Changing Parsec to use Data.Text

### PR DESCRIPTION
**Pull request by christinem**

### CODE REVIEW

File: app/WebParsing/ParsecCombinators.hs [hunk 1]
```diff
@@ -12,23 +12,25 @@ module WebParsing.ParsecCombinators
 import qualified Text.Parsec as P
 import Text.Parsec ((<|>))
 import qualified Data.Text as T
-import Text.Parsec.String (Parser)
+import Text.Parsec.Text (Parser)
 import Database.Tables
+import Control.Monad
```
Comment (david-yz-liu) [inline app/WebParsing/ParsecCombinators.hs]: Specify which functions are being imported here.

File: app/WebParsing/ParsecCombinators.hs [hunk 2]
```diff
@@ -37,86 +39,94 @@ generalCategoryParser firstCourse postCode = do
...
+   parsed <- P.many1 P.letter
+   return $ T.pack parsed
 
-getDepartmentName :: Parser String
-getDepartmentName =
-    P.try (parseUntil (P.try (P.lookAhead (P.string " Specialist")) <|>
-                        P.try (P.lookAhead (P.string " Major")) <|>
-                        P.try (P.lookAhead (P.string " Minor"))))
+getDepartmentName :: Parser T.Text
+getDepartmentName =
```
Comment (david-yz-liu) [inline app/WebParsing/ParsecCombinators.hs]: There's trailing space here to remove.

File: app/WebParsing/ParsecCombinators.hs [hunk 3]
```diff
@@ -37,86 +39,94 @@ generalCategoryParser firstCourse postCode = do
...
     nextChar <- P.anyChar
     return left
 
-parseUpToSeparator :: Parser String
+parseUpToSeparator :: Parser T.Text
 parseUpToSeparator = parseUntil (P.notFollowedBy (P.noneOf ";\r\n"))
 
+text :: T.Text -> Parser T.Text
+text someText = do
+    parsed <- mapM P.char (T.unpack someText)
```
Comment (david-yz-liu) [inline app/WebParsing/ParsecCombinators.hs]: Shouldn't `P.char` also work on Text?

File: app/WebParsing/PostParser.hs [hunk 4]
```diff
@@ -47,32 +47,32 @@ addPostToDatabase :: [Tag String] -> SqlPersistM ()
 addPostToDatabase tags = do
     let postCode = T.pack (fromAttrib "name" ((take 1 $ filter (isTagOpenName "a") tags) !! 0))
         liPartitions = partitions isLiTag tags
-        prereqs = map getCourseFromTag $ map (fromAttrib "href") $ filter isCourseTag tags
+        prereqs = map getCourseFromTag $ map T.pack $ map (fromAttrib "href") $ filter isCourseTag tags
```
Comment (david-yz-liu) [inline app/WebParsing/PostParser.hs]: This is better expressed using composition: `(T.pack . fromAttrib "href")`

File: app/WebParsing/PostParser.hs [hunk 5]
```diff
@@ -81,16 +81,16 @@ categoryParser tags firstCourse postCode liPartitions = do
         Left message -> do
             return ()
     where
-        parsed = case liPartitions of
-            [] -> P.parse (generalCategoryParser firstCourse postCode) "Failed." (innerText tags)
+        parsed = case liPartitions of
```
Comment (david-yz-liu) [inline app/WebParsing/PostParser.hs]: More whitespace here.

**Comment by christinem** [comment]
@david-yz-liu Comments have been addressed.

**Pull request merged**